src/utils/slpParser.ts

```diff
@@ -266,3 +282,42 @@ export class SlpParser extends EventEmitter {
     }
   }
 }
+
+class RollbackCounter {
+  private rollbackFrames: RollbackFramesType = {};
+  private rollbackFrameCount = 0;
+  private rollbackPlayerIdx: number | null = null; // for keeping track of rollbacks by following a single player
+  private lastFrameWasRollback = false;
+  private currentRollbackLength = 0;
+  private rollbackLengths: number[] = [];
+
+  public checkIfRollbackFrame(currentFrame: FrameEntryType | undefined, playerIdx: number) {
+    if (this.rollbackPlayerIdx === null) {
+      // we only want to follow a single player to avoid double counting. So we use whoever is on first.
+      this.rollbackPlayerIdx = playerIdx;
+    } else if (this.rollbackPlayerIdx !== playerIdx) {
+      return;
+    }
+
+    if (currentFrame) {
+      // frame already exists for currentFrameNumber so we must be rolling back
+      if (this.rollbackFrames[currentFrame.frame]) {
+        this.rollbackFrames[currentFrame.frame].push(currentFrame);
+      } else {
+        this.rollbackFrames[currentFrame.frame] = [currentFrame];
+      }
+      this.rollbackFrameCount++;
+      this.currentRollbackLength++;
+      this.lastFrameWasRollback = true;
+    } else if (this.lastFrameWasRollback) {
+      this.rollbackLengths.push(this.currentRollbackLength);
+      this.currentRollbackLength = 0;
+      this.lastFrameWasRollback = false;
+    }
+    return this.lastFrameWasRollback;
+  }
+
+  public get(): RollbackFrames {
```

Comment:
I'm not sure i like the naming of `.get()`. `rollbackCounter.get()` sounds like to me it would return just the count. I think if we had like `.getFrames()`, `.getCount()`, and `.getLengths()` as 3 separate methods, it would be better, that way we don't care if the shape of `RollbackFrames` changes.